### PR TITLE
fix: removed strict template errors from v2 linter

### LINKED ISSUE #6705: lint undefined keys and "and"
defaults:
    image_tag: ~

tag: '{{- if and (hasKey .Values.defaults "image_tag") (.Values.defaults.image_tag){{.Values.defaults.image_tag

The above line, with definition in values, does not pass lint strict because if complains that that the key ".Values.defaults.image_tag" is not defined.

1. It is defined as nil
2. There is a check in the first and term to make sure it is defined before it is referenced

Output of `helm version`:

2.14.3

Output of `kubectl version`:

Cloud Provider/Platform (AKS, GKE, Minikube etc.):

### PR DESCRIPTION
**Pull request by jeff-knurek**
Signed-off-by: Jeff Knurek <knurek.stuff@gmail.com>

**What this PR does / why we need it**:
closes #6705 

**Special notes for your reviewer**:
followed similar solution as in v3 (#8017) ~~but also cleared out the need for passing the `strict` bool around to different functions~~
* also added a test to the lintCmd to confirm the `--strict` flag functions as expected

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility _(some charts that did not pass `helm lint --strict` will now pass.)_

### CODE REVIEW

**Review by bacongobbler — APPROVED** [review]
LGTM. Will need another maintainer to sign off on this before we can merge.

File: pkg/lint/rules/template.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ import (
 )
 
 // Templates lints the templates in the Linter.
-func Templates(linter *support.Linter, values []byte, namespace string, strict bool) {
+func Templates(linter *support.Linter, values []byte, namespace string) {
```
Comment (hickeyma) [inline pkg/lint/rules/template.go:35]: Is this a backwards compatibility change? If so then the change should only be the removal of line as you did below:

```yaml
if strict {	
    e.Strict = true	
}
```

Maybe, just make changes similar to #8017?

**Review by hickeyma — CHANGES_REQUESTED** [review]
Thanks for pushing the PR @jeff-knurek. Comment inline about backwards compatibility.

File: pkg/lint/rules/template.go (hunk 1 above)
Comment (technosophos) [reply pkg/lint/rules/template.go:35]: I agree. It is okay to leave a no-op param in the name of preserving API compatibility.

Comment (jeff-knurek) [reply pkg/lint/rules/template.go:35]: oh, you're right, I wasn't thinking about any external usage. I agree that it does make more sense to keep the param. 

however, is the change really still "backwards compatible" if someone using the library function with the current logic, and is still expecting `e.Strict = true` to be set? I mean to say that such a change _should_ break the api. but the benefits for doing so in `v2.x` don't outweigh the negative impact of breaking the api, so i'll edit the PR to keep the param.

**Comment by jeff-knurek** [comment]
test failure unrelated to change:
```
FAIL	k8s.io/helm/cmd/tiller [build failed]
```

**Review by hickeyma — CHANGES_REQUESTED** [review]
Thanks for the updates @jeff-knurek. It's nearly there. There is just some cleanup of unnecessary test files as follows:

- `cmd/helm/testdata/testcharts/bad.name/.helmignore`
- `cmd/helm/testdata/testcharts/bad.name/templates/*` (you can leave the `templates` dir)

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @jeff-knurek.

**Review by technosophos — APPROVED** [review]
This looks great. Thanks for getting this done before the code freeze deadline

**Pull request merged**